apis/v1alpha1/shared_types.go

```diff
@@ -175,6 +189,33 @@ type AdminNetworkPolicyEgressPeer struct {
 	// <network-policy-api:experimental>
 	// +optional
 	Nodes *metav1.LabelSelector `json:"nodes,omitempty"`
+	// Networks defines a way to select peers via CIDR blocks (both v4 & v6).
+	// This is intended for representing entities that live outside the cluster,
+	// which can't be selected by pods, namespaces and nodes peers, but note
+	// that cluster-internal traffic will be checked against the rule as
+	// well. So if you Allow or Deny traffic to `"0.0.0.0/0"`, that will allow
+	// or deny all IPv4 pod-to-pod traffic as well. If you don't want that,
+	// add a rule that Passes all pod traffic before the Networks rule.
+	//
+	// Do not provide serviceCIDR range in `networks` peer. This is NOT supported.
+	// While 0.0.0.0/0 will match on the serviceVIP as well, it does not make much sense
+	// anyways in that case because policies are expected to be applied after the serviceVIP
+	// (clusterIP, externalIP, loadbalancer.Ingress.VIP) is rewritten to the actual backend IPs.
+	// So policy match finally gets applied on the endpointIPs not serviceVIPs.
```

Comment:
(ignoring my earlier comments and just trying to tighten the existing wording here up)

```suggestion
	// Note that because policies are applied after Service VIPs (clusterIPs, externalIPs,
	// load balancer IPs) are rewritten to endpoint IPs, a Networks selector cannot match
	// such a VIP. For example, a Networks selector that denies traffic to the entire
	// service CIDR will not actually block any service traffic.
```

Reply:
yeah I like your wording here!!! :D infact I was about to ping you to help me with the wordings :D its an art! thanks Dan for your expert review here, much appreciated!